Dockerfile

```diff
@@ -1,37 +1,77 @@
-FROM kuzzleio/base
+################################################################################
+# Plugin development image
+################################################################################
+FROM debian:stretch-slim as plugin-dev
 
 LABEL io.kuzzle.vendor="Kuzzle <support@kuzzle.io>"
-LABEL description="Power your web, mobile & iot apps with the Kuzzle backend"
+LABEL description="Develop new plugin or protocol for Kuzzle with ease"
 
-COPY ./ /var/app/
-COPY ./docker-compose/scripts/run.sh /usr/local/bin/kuzzle
+ENV NODE_VERSION=8.9.0
+ENV PATH=/opt/node-v$NODE_VERSION-linux-x64/bin:$PATH
+
+ARG kuzzle_tag
+
+ADD https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz /tmp/
+
+RUN  set -x \
+  \
+  && apt-get update && apt-get install -y \
+       bash-completion \
+       build-essential \
+       curl \
+       g++ \
+       gdb \
+       git \
+       python \
+       libfontconfig \
+       wget \
+  && tar xf /tmp/node-v$NODE_VERSION-linux-x64.tar.gz -C /opt/ \
+  && rm /tmp/node-v$NODE_VERSION-linux-x64.tar.gz \
+  && mkdir -p /var/app \
+  && npm install -g npm \
+  && npm set progress=false \
+  && npm install -g \
+    pm2 \
+  && echo "" > /opt/node-v$NODE_VERSION-linux-x64/lib/node_modules/pm2/lib/keymetrics \
+  && rm -rf /var/lib/apt/lists/* \
+  && echo "alias ll=\"ls -lahF --color\"" >> ~/.bashrc
+
+RUN  git clone --depth 1 --branch $kuzzle_tag https://github.com/kuzzleio/kuzzle /var/app \
+  && git clone --depth 1 https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local /var/app/plugins/available/kuzzle-plugin-auth-passport-local \
+  && git clone --depth 1 https://github.com/kuzzleio/kuzzle-plugin-logger /var/app/plugins/available/kuzzle-plugin-logger
 
 WORKDIR /var/app
 
+RUN  npm install --unsafe-perm \
+  && npm rebuild all --unsafe-perm \
+  && for plugin in plugins/enabled/*; do cd "$plugin"; npm install --unsafe; cd /var/app; done
+
+################################################################################
+# Production image
+################################################################################
+FROM debian:stretch-slim as kuzzle
+
+LABEL io.kuzzle.vendor="Kuzzle <support@kuzzle.io>"
+LABEL description="Run your Kuzzle backend in production mode"
+
+ENV NODE_VERSION=8.9.0
+ENV NODE_ENV=production
+ENV PATH=/opt/node-v$NODE_VERSION-linux-x64/bin:$PATH
+
 RUN  apt-get update \
   && apt-get install -y \
-    build-essential \
-    curl \
-    git \
-    g++ \
-    gdb \
-    python \
-  \
-  && npm install --unsafe-perm -g pm2 \
-  && npm install --unsafe-perm \
-  && npm rebuild all --unsafe-perm \
-  && for plugin in plugins/enabled/*; do cd "$plugin"; npm install --unsafe-perm; cd /var/app; done \
-  \
-  && chmod a+x /usr/local/bin/kuzzle \
-  && chmod a+x /var/app/docker-compose/scripts/docker-entrypoint.sh \
-  \
-  && apt-get clean \
-  && apt-get remove -y \
-    build-essential \
-    g++ \
-    python \
-  && apt-get autoremove -y \
+       curl \
```

Comment:
I'm not sure that we need `curl` in our production image but some of our script (the one that is waiting Elasticsearch to be up for example) need `curl`.

Reply:
Since we already install `wget`, maybe we could use it to remove curl... in a later PR.

Reply:
`wget` is not present on the production image. We don't install any packet except curl